packages/angular_devkit/build_optimizer/src/transforms/class-fold.ts

```diff
@@ -61,8 +67,49 @@ export function getFoldFileTransformer(program: ts.Program): ts.TransformerFacto
             newBody,
           );
 
+          // Update remaining classes and statements.
+          statementsToRemove.push(...clazz.statements.map(st => st.expressionStatement));
+          classes = classes.filter(cl => cl != clazz);
+
           // Replace node with modified one.
-          return ts.visitEachChild(newNode, visitor, context);
+          return newNode;
+        }
+
+        // Check if node is a ES2015 class to replace with a pure IIFE.
+        clazz = classes.find((cl) => !cl.function && cl.declaration === node);
+        if (clazz) {
+          // There doesn't seem to be a way to make a statement from a class declaration.
+          // But it seems to work if you just jam it in there.
+          const classStatement = clazz.declaration as {} as ts.ExpressionStatement;
```

Comment:
A `ClassDeclaration` is already a statement.  The casting shouldn't be necessary.

Reply:
Fixed (the right cast was to `ts.ClassDeclaration`, since it could be either that or `ts.VariableDeclaration`).